_pytest/warnings.py

```diff
@@ -72,8 +72,8 @@ def catch_warnings_for_item(item):
             unicode_warning = False
 
             if compat._PY2 and any(isinstance(m, compat.UNICODE_TYPES) for m in warn_msg.args):
-                new_args = [compat.safe_str(m) for m in warn_msg.args]
```

Comment:
We didn't realize at the time, but this is incorrect, `safe_str` will return `utf-8` encoded `bytes`, which is not `ascii`.